doozer/doozerlib/cli/scan_osh.py

```diff
@@ -0,0 +1,114 @@
+import click
+import koji
+from doozerlib.cli import cli, click_coroutine, pass_runtime
+from doozerlib.runtime import Runtime
+from doozerlib.exectools import fire_and_forget
+from doozerlib.constants import BREWHUB_URL
+from doozerlib.util import cprint
+
+
+class ScanOshCli:
+    def __init__(self, runtime: Runtime, brew_tags: list, last_brew_event):
+        self.runtime = runtime
+        self.brew_tags = brew_tags
+        self.last_brew_event = last_brew_event
+
+        # Initialize runtime
+        self.runtime.initialize(clone_distgits=False, no_group=True)
+
+    @staticmethod
+    async def get_tagged_latest(tag):
+        """
+        Returns the latest RPMs and builds tagged in to the candidate tag received as input
+        """
+        session = koji.ClientSession(BREWHUB_URL)
+        try:
+            latest_tagged = session.listTagged(tag=tag, latest=True)
+            if latest_tagged:
+                return latest_tagged
+        except Exception:
+            raise
+
+    @staticmethod
+    async def get_tagged_all(tag):
```

Comment:
Why is this async?

Reply:
ahh since its an API call I thought async will be faster? unless I'm mistaken

Reply:
Unless we're creating multiple coroutines then this doesn't make a difference

Reply:
Also take a look at different koji client implementations, there's "build_retrying_koji_client" that we can use for auto retry. If you want to create multiple threads, then "shared_koji_client_session" is what we should use

Reply:
Understood. Updated in https://github.com/openshift-eng/art-tools/pull/51/commits/35128eb17ea67bb154c2c093d91a370131e52f55